### ISSUE TITLE
Can't run tests

### ISSUE THREAD

**Issue by mperham** [mperham/sidekiq.cr#105]
I installed Crystal 1.0 but I can't run the specs right now.

```
❯ shards install
Resolving dependencies
Fetching https://github.com/stefanwille/crystal-redis.git
Fetching https://github.com/ysbaddaden/pool.git
Fetching https://github.com/kemalcr/kemal.git
Fetching https://github.com/kemalcr/kemal-session.git
Fetching https://github.com/kemalcr/kemal-csrf.git
Fetching https://github.com/schovi/baked_file_system.git
Fetching https://github.com/crystal-community/timecop.cr.git
Fetching https://github.com/crystal-ameba/ameba.git
Unable to satisfy the following requirements:

- `crystal (< 1.0.0)` required by `redis 2.6.0`
- `crystal (>= 0.36.0)` required by `kemal 1.0.0`
- `crystal (>= 0.36.0)` required by `kemal-session 1.0.0`
- `crystal (>= 1.0)` required by `kemal-csrf 0.8.0+git.commit.abaee30f1e8e28d896ed6bf5267a530b5d6ca638`
- `crystal (~> 0.25, >= 0.25.0)` required by `baked_file_system 0.9.8+git.commit.7183bfde8d86a976a6912f582b51cbd1783456af`
- `crystal (>= 0.33.0)` required by `timecop 0.4.1`
- `crystal (>= 0.35.0)` required by `ameba 0.14.2`
- `crystal (< 1.0.0)` required by `pool 0.2.3`
Failed to resolve dependencies, try updating incompatible shards or use --ignore-crystal-version as a workaround if no update is available.
```

Let's get this working and update the wiki with the latest details.

https://github.com/mperham/sidekiq.cr/wiki/Getting-Started

As soon as I can verify the code, I'll tag `v0.8.0`.

**Comment by hugopl** [comment]
Some dependencies didn't release yet a version targeting crystal 1.0.0, but they work with crystal 1.0.0, so the need of use `shards install --ignore-crystal-version` instead of just `shards install`

A explanation can be found at: https://crystal-lang.org/2020/08/20/preparing-our-shards-for-crystal-1.0.html

**Comment by hugopl** [comment]
I added the `--ignore-crystal-version` to https://github.com/mperham/sidekiq.cr/wiki/Getting-Started

I also removed the `branch: master`, so instead of get the code from master, shards get the last tag.

BTW the last tag is `v0.6.1`, so the next would be `v0.7.0`, not `v0.8.0`.

**Comment by mperham** [comment]
Ok, I'll use 0.7.0.

Is there a way to easily remove any runtime dependencies? Does Crystal stdlib have anything like `baked_file_system`, for example?

Do we need to make any changes to improve Sidekiq's multithreaded support?

I don't follow Crystal anymore so I'll be relying on you and the rest of the community to suggest improvements.

**Comment by hugopl** [comment]
> Is there a way to easily remove any runtime dependencies? Does Crystal stdlib have anything like `baked_file_system`, for example?

They are compile-time dependencies, there's nothing like `baked_file_system` on stdlib that I know.
 
> Do we need to make any changes to improve Sidekiq's multithreaded support?

To to let the Fibers be spawn in different "real OS threads" the user should compile the code with `-Dpreview_mt`. But this is a general Crystal issue, not a sidekiq specific issue.

These issues with shards not having released a version compatible with Crystal 1.0.0 will be solved with the time.

Since the last release was ages ago, I think we are good to release a `v0.7.0`.

**Comment by mperham** [comment]
Sorry, yes, compile test but necessary for Sidekiq.cr to run. I know Go 1.16 added embedded filesystem support and was hoping Crystal would have something similar so we could remove the dependency.

Will tag now.
